### PR TITLE
Add initial changes and tests for report verification

### PR DESCRIPTION
**Pull request by abaiken**

### CODE REVIEW

File: yupana/processor/kafka_msg_handler.py [hunk 1]
```diff
@@ -146,9 +146,36 @@ def verify_report_details(report_contents):
         LOG.error(message % 'system_fingerprints')
         status = FAILURE_CONFIRM_STATUS
 
+    if fingerprints and report_platform_id:
+        verified_fingerprints = verify_report_fingerprints(fingerprints, report_platform_id)
+        if verified_fingerprints:
+            # FIXME: Cody add stuff here
```
Comment (kholdaway) [inline yupana/processor/kafka_msg_handler.py]: You could add a function instead of a fixme.  Have the funciton just use a `pass`

**Review by kholdaway — CHANGES_REQUESTED** [review]

File: yupana/processor/kafka_msg_handler.py (hunk 1 above)
Comment (abaiken) [reply yupana/processor/kafka_msg_handler.py]: I think that this has the same functionality but if you have a strong opinion, I can change this.

File: yupana/processor/kafka_msg_handler.py [hunk 2]
```diff
@@ -146,9 +146,36 @@ def verify_report_details(report_contents):
...
+    """Verify that report fingerprints contain canonical facts."""
+    canonical_facts = ['insights_client_id', 'bios_uuid', 'ip_addresses', 'mac_addresses',
+                       'vm_uuid', 'etc_machine_id', 'subscription_manager_id']
+    message = 'report_platform_id: "%s"| Removing invalid fingerprint. No canonical facts: %s.'
+    verified_fingerprints = []
+    for fingerprint in fingerprints:
+        found_facts = False
+        for fact in canonical_facts:
+            if fingerprint.get(fact):
+                found_facts = True
```
Comment (kholdaway) [inline yupana/processor/kafka_msg_handler.py]: You should break out of the loop here.

File: yupana/processor/kafka_msg_handler.py (hunk 1 above)
Comment (abaiken) [reply yupana/processor/kafka_msg_handler.py]: Fixed

File: yupana/processor/kafka_msg_handler.py (hunk 2 above)
Comment (abaiken) [reply yupana/processor/kafka_msg_handler.py]: Fixed

**Review by myersCody — APPROVED** [review]

**Review by kholdaway — APPROVED** [review]

**Pull request merged**